master/usage/configuration/ip-in-ip.md

```diff
@@ -0,0 +1,112 @@
+apiVersion: v1
+kind: ipPool
+metadata:
+  cidr: 192.168.0.0/16
+spec:
+  ipip:
+    enabled: true
+    mode: always
+EOF
+```
+
+> Note that the default value for `mode` is `always`, and therefore may be omitted
+> from the request, but is shown above for clarity.
+
+### Configuring cross-subnet IP-in-IP
+
+If IP-in-IP encapsulation is required for L3-hops, but not required within
+the L2 subnet, then set the IP-in-IP mode to `cross-subnet`.  With this mode, Calico
+will route using IP-in-IP encapsulation for inter-workload traffic to workloads in 
+the configured pool *only* if the  destination host is in a different subnet to the 
+originating host.
+
+This avoids unneccessary encapsulation within the same subnet.
+
+The following `calicoctl` example will create or modify an IPv4 pool with 
+CIDR 192.168.0.0/16 to use IP-in-IP with mode `cross-subnet`:
+
+
+```
+$ calicoctl apply -f - << EOF
+apiVersion: v1
+kind: ipPool
+metadata:
+  cidr: 192.168.0.0/16
+spec:
+  ipip:
+    enabled: true
+    mode: cross-subnet
+EOF
+```
+
+> **Note** 
+>
+> The `cross-subnet` mode option requires each Calico node to be configured
+> with the IP address and subnet of the host.  However, the subnet configuration
+> was only introduced in Calico v2.1.  If any nodes in your deployment were originally
+> created with an older version of Calico, or if you if you are unsure whether 
+> your deployment is configured correctly, follow the steps in 
+> [Upgrading from pre-v2.1](#upgrading-from-pre-v21) before enabling "cross-subnet" IPIP.
+
+#### Upgrading from pre-v2.1
+
+If you are planning to use cross-subnet IPIP, your entire deployment must be running with
+Calico v2.1 or higher.  See [releases page]({{site.baseurl}}/{{page.version}}/releases) 
+for details on the component versions for each release.
+
+Upgrade your deployment to use the latest Calico versions - the process for this
+will be dependent on your orchestration system (if using one). 
+
+Prior to Calico v2.1, the subnet information was not detected and stored on the 
+node configuration.  Thus, if you have calico/node instances that were deployed 
+prior to v2.1, the node configuration may need updating to fix the host subnet.
+The subnet configuration must be set correctly for each node before `cross-subnet`
+IPIP mode is enabled.
+
+You can verify which of your nodes is correctly configured using calicoctl.
+
+Run `calicoctl get nodes --output=wide` to check the configuration.  e.g.
+
+```
+$ calicoctl get nodes --output=wide
+NAME    ASN       IPV4           IPV6   
+node1   (64512)   10.0.2.15/24          
+node2   (64512)   10.0.2.10/32          
+```
+
+In this example, node1 has the correct subnet information whereas node2 needs
+to be fixed.
+
+The subnet configuration may be fixed in a few different ways depending on how 
+you have deployed your calico/node containers.  This is discussed in the
+[Configuring a node guide]({{site.baseurl}}/{{page.version}}/usage/configuration/node).
```

Comment:
Let's make this link consistent with the title of the guide - "Configuring a Node IP Address and Subnet"